src/RevExt.cc

```diff
@@ -11,78 +11,31 @@
 #include "../include/RevExt.h"
 using namespace SST::RevCPU;
 
-#define TRACE_EXECUTE 0
-
-bool RevExt::Execute(unsigned Inst, RevInst payload, uint16_t HartID){
-
-  // ensure that the target instruction is within scope
-  if( Inst >= table.size() ){
-    output->fatal(CALL_INFO, -1,
-                  "Error: instruction at index=%u does not exist in extension=%s",
-                  Inst,
-                  name.c_str());
-  }
-
+bool RevExt::Execute(unsigned Inst, const RevInst& payload, uint16_t HartID, RevRegFile* regFile){
   bool (*func)(RevFeature *,
                RevRegFile *,
                RevMem *,
                RevInst) = nullptr;
-  if( payload.compressed ){
-#if TRACE_EXECUTE
-    if( feature->IsRV32() ){
-      std::cout << "EXECUTING COMPRESSED INSTRUCTION: " << ctable[Inst].mnemonic
-                << " @ 0x" << std::hex << regFile[threadID].RV32_PC << std::dec
-                << "; instSize = " << payload.instSize << std::endl;
+  try{
+    if( payload.compressed ){
+      // this is a compressed instruction, grab the compressed trampoline function
+      func = ctable.at(Inst).func;
     }else{
-      std::cout << "EXECUTING COMPRESSED INSTRUCTION: " << ctable[Inst].mnemonic
-                << " @ 0x" << std::hex << regFile[threadID].RV64_PC << std::dec
-                << "; instSize = " << payload.instSize << std::endl;
+      // retrieve the function pointer for this instruction
+      func = table.at(Inst).func;
     }
-#endif
-    // this is a compressed instruction, grab the compressed trampoline function
-    func = ctable[Inst].func;
-  }else{
-#if TRACE_EXECUTE
-    if( feature->IsRV32() ){
-      std::cout << "EXECUTING INSTRUCTION: " << table[Inst].mnemonic
-                << " @ 0x" << std::hex << regFile[threadID].RV32_PC << std::dec
-                << "; instSize = " << payload.instSize << std::endl;
-    }else{
-      std::cout << "EXECUTING INSTRUCTION: " << table[Inst].mnemonic
-                << " @ 0x" << std::hex << regFile[threadID].RV64_PC << std::dec
-                << "; instSize = " << payload.instSize << std::endl;
-    }
-#endif
-    // retrieve the function pointer for this instruction
-    func = table[Inst].func;
+  } catch(...) {
+    output->fatal(CALL_INFO, -1,
+                  "Error: instruction at index=%u does not exist in extension=%s",
+                  Inst,
+                  name.data());
+    throw;
   }
```

Comment:
My limited knowledge of c++17 says `try-catch` is abysmal for performance. Is the reason this is acceptable here because it's only if `TRACE_EXECUTE`?

Reply:
No, `TRACE_EXECUTE` was removed.

I can remove the `try`-`catch` block too. The original code was very cumbersome by doing its own bounds checking, while I simplified it to use `.at(i)` instead of `[i]`, and catch if `at()` exceptions.

With this and the removal of the uncompilable `TRACE_EXECUTE` code, `RevExt::Execute` is much smaller.

I've heard performance is slow when exceptions occur, but as long as they don't occur, it should be fast. I can change it back, but I'd use a little different syntax.